tensorflow_similarity/callbacks.py

```diff
@@ -142,14 +203,48 @@ def on_epoch_end(self, epoch: int, logs: dict = None):
             distance_thresholds=self.distance_thresholds,
         )
 
+        if self.split_validation:
+            unknown_results = _compute_classification_metrics(
+                queries=self.queries_unknown,
+                query_labels=self.query_labels_unknown,
+                model=self.model,
+                evaluator=self.evaluator,
+                metrics=self.metrics,
+                k=self.k,
+                matcher=self.matcher,
+                distance_thresholds=self.distance_thresholds,
+            )
+
         mstr = []
-        for metric_name, vals in results.items():
-            float_val = vals[0]
-            logs[metric_name] = float_val
-            mstr.append(f"{metric_name}: {float_val:.4f}")
-            if self.tb_writer:
-                with self.tb_writer.as_default():
-                    tf.summary.scalar(metric_name, float_val, step=epoch)
+        if self.split_validation:
+            for metric_name, vals in known_results.items():
```

Comment:
I feel like these are all so similar that we could make a simple helper method for it. Maybe something like:

```
def unpack_results(results: Dict[str, np.ndarray], name_suffix: str = '') -> List[str]:
    ...
```